kyaml/kio/byteio_reader.go

```diff
@@ -133,6 +134,8 @@ type ByteReader struct {
 
 var _ Reader = &ByteReader{}
 
+var yamlSeparatorRegexp = regexp.MustCompile(`\n---.*\n`)
```

Comment:
what's the motivation for declaring this outside of the function?

Reply:
It is to avoid compiling the same regexp multiple times if `Read()` is invoked multiple times.

I've seen it done this way frequently in k/k and have taken this approach myself before too, but to be honest I don't know how much of an impact it would really have on things.

I'll move it inside the function and we can deal with it later if it becomes an issue.